tests/ert/ui_tests/cli/test_cli.py

```diff
@@ -557,6 +557,79 @@ def test_that_stop_on_fail_workflow_jobs_stop_ert(
         run_cli(TEST_RUN_MODE, "--disable-monitor", "poly.ert")
 
 
+@pytest.mark.usefixtures("copy_poly_case")
+def test_that_pre_post_experiment_hook_works(
+    monkeypatch,
+):
+    monkeypatch.setattr(_ert.threading, "_can_raise", False)
+
+    # The executable
+    with open("hello_post_exp.sh", "w", encoding="utf-8") as f:
+        f.write(
+            dedent("""#!/bin/bash
+                echo "just sending regards" > from_post_experiment.txt
+        """)
+        )
+    os.chmod("hello_post_exp.sh", 0o755)
+
+    # The workflow job
+    with open("SAY_HELLO_POST_EXP", "w", encoding="utf-8") as s:
+        s.write("""
+               INTERNAL False
+               EXECUTABLE hello_post_exp.sh
+           """)
+
+    # The workflow
+    with open("SAY_HELLO_POST_EXP.wf", "w", encoding="utf-8") as s:
+        s.write("""dump_final_ensemble_id""")
+
+    # The executable
+    with open("hello_pre_exp.sh", "w", encoding="utf-8") as f:
+        f.write(
+            dedent("""#!/bin/bash
+                echo "first" > from_pre_experiment.txt
+        """)
+        )
+    os.chmod("hello_pre_exp.sh", 0o755)
+
+    # The workflow job
+    with open("SAY_HELLO_PRE_EXP", "w", encoding="utf-8") as s:
+        s.write("""
+               INTERNAL False
+               EXECUTABLE hello_pre_exp.sh
+           """)
+
+    # The workflow
+    with open("SAY_HELLO_PRE_EXP.wf", "w", encoding="utf-8") as s:
+        s.write("""dump_first_ensemble_id""")
+
+    with open("poly.ert", mode="a", encoding="utf-8") as fh:
+        fh.write(
+            dedent(
+                """
+                    NUM_REALIZATIONS 2
+
+                    LOAD_WORKFLOW_JOB SAY_HELLO_POST_EXP dump_final_ensemble_id
+                    LOAD_WORKFLOW SAY_HELLO_POST_EXP.wf POST_EXPERIMENT_DUMP
+                    HOOK_WORKFLOW POST_EXPERIMENT_DUMP POST_EXPERIMENT
+
+                    LOAD_WORKFLOW_JOB SAY_HELLO_PRE_EXP dump_first_ensemble_id
+                    LOAD_WORKFLOW SAY_HELLO_PRE_EXP.wf PRE_EXPERIMENT_DUMP
+                    HOOK_WORKFLOW PRE_EXPERIMENT_DUMP PRE_EXPERIMENT
+                """
+            )
+        )
+
+    run_cli(ITERATIVE_ENSEMBLE_SMOOTHER_MODE, "--disable-monitor", "poly.ert")
+
+    assert (Path(os.getcwd()) / "from_pre_experiment.txt").read_text(
+        "utf-8"
+    ) == "first\n"
+    assert (Path(os.getcwd()) / "from_post_experiment.txt").read_text(
+        "utf-8"
+    ) == "just sending regards\n"
+
+
```

Comment:
For example:

```suggestion
@pytest.mark.parametrize("hook", ["PRE", "POST"])
def test_that_pre_post_experiment_hook_works(
    monkeypatch, tmp_path, hook, capsys
):
    monkeypatch.chdir(tmp_path)
    monkeypatch.setattr(_ert.threading, "_can_raise", False)

    # The executable
    with open("hello.sh", "w", encoding="utf-8") as f:
        f.write(
            dedent("""#!/bin/bash
                echo "just sending regards"
        """)
        )
    os.chmod("hello.sh", 0o755)

    # The workflow job
    with open("SAY_HELLO", "w", encoding="utf-8") as s:
        s.write("""
               INTERNAL False
               EXECUTABLE hello.sh
           """)

    # The workflow
    with open("SAY_HELLO_CONFIG", "w", encoding="utf-8") as s:
        s.write("""SAY_HELLO""")

    with open("poly.ert", mode="a", encoding="utf-8") as fh:
        fh.write(
            dedent(
                f"""
                    NUM_REALIZATIONS 2
                    LOAD_WORKFLOW_JOB SAY_HELLO
                    LOAD_WORKFLOW SAY_HELLO_CONFIG
                    HOOK_WORKFLOW SAY_HELLO_CONFIG {hook.upper()}_EXPERIMENT
                """
            )
        )


    run_cli(ENSEMBLE_EXPERIMENT_MODE, "--disable-monitor", "poly.ert")
    captured = capsys.readouterr()
    assert "just sending regards" in captured.out

```

could also look at: `test_model_hook_order`, so it can be tested for all relevant experiments, and not just ies